src/passwd.c

```diff
@@ -1034,22 +977,15 @@ int main (int argc, char **argv)
 #ifdef WITH_SELINUX
 	/* only do this check when getuid()==0 because it's a pre-condition for
 	   changing a password without entering the old one */
-	if ((is_selinux_enabled() > 0) && (getuid() == 0) &&
-	    (check_selinux_access (name, pw->pw_uid, PASSWD__PASSWD) != 0)) {
-		security_context_t user_context = NULL;
-		const char *user = "Unknown user context";
-		if (getprevcon (&user_context) == 0) {
-			user = user_context; /* FIXME: use context_user_get? */
-		}
+	current_uid = getuid() ;
+	if ((0 == current_uid) &&
```

Comment:
We should probably be checking the "amroot" variable used elsewhere for consistency.  Otherwise, if the amroot logic changes later, we won't match.

Reply:
The result of `0 == current_uid` (where `current_uid = getuid()`) should be the same as the value `amroot`. So simplify the whole condition to (ignoring the shortcut logic for now)?
```
if (amroot && (check_selinux_permit_shortcut ("passwd", name, pw->pw_uid) != 0)) {
```

Reply:
Correct.  That way if someone later alters the amroot logic we'll pick up the change automatically, and keep in sync.  We want to do this check iff it was possible to change the password without entering the old one.

Reply:
changed